lib/screens/v2/widget_instance/reconstructed_alert.ex

```diff
@@ -72,183 +139,351 @@ defmodule Screens.V2.WidgetInstance.ReconstructedAlert do
     end
   end
 
-  defp get_route_pills(routes) do
-    routes
+  defp get_route_pills(t, location \\ nil)
+
+  defp get_route_pills(t, nil) do
+    affected_routes = LocalizedAlert.informed_subway_routes(t)
+
+    affected_routes
     |> Enum.group_by(fn
       "Green" <> _ -> "Green"
       route -> route
     end)
     |> Enum.map(
-      &RoutePill.serialize_route_for_reconstructed_alert(&1, %{large: length(routes) == 1})
+      &RoutePill.serialize_route_for_reconstructed_alert(&1, %{
+        large: length(affected_routes) == 1
+      })
     )
   end
 
-  def takeover_alert?(
-        %{
-          screen: %Screen{app_id: :pre_fare_v2},
-          is_terminal_station: is_terminal_station,
-          alert: alert
-        } = t
-      ) do
+  defp get_route_pills(%__MODULE__{alert: alert} = t, location) do
+    informed_entities = Alert.informed_entities(alert)
+    routes_at_stop = LocalizedAlert.active_routes_at_stop(t)
+
+    informed_entities
+    |> Enum.filter(&(&1.route_type in [0, 1] and &1.route in routes_at_stop))
+    |> Enum.group_by(fn %{route: route} -> route end)
+    |> Enum.map(fn
+      {route_id, _} ->
+        headsign = get_destination(t, location, route_id)
+
+        if is_nil(headsign) do
+          route_id |> String.first() |> String.downcase() |> Kernel.<>("l")
+        else
+          format_for_svg_name(headsign)
+        end
+    end)
+    |> Enum.uniq()
+  end
+
+  defp format_for_svg_name("Ashmont/Braintree"),
+    do: [Map.get(@headsign_svg_map, "Ashmont"), Map.get(@headsign_svg_map, "Braintree")]
+
+  defp format_for_svg_name(headsign), do: Map.get(@headsign_svg_map, headsign)
+
+  defp format_cause(:unknown), do: nil
+  defp format_cause(cause), do: cause |> to_string() |> String.replace("_", " ")
```

Comment:
The problem with that function is the `due to ...`. The cause in the updated alerts is part of the footer where we just say `Cause: ...`. So the client just needs the formatted cause instead of the `due to ...`.

Reply:
Ok, all good as-is then.